src/torchcrf/__init__.py

```diff
@@ -244,54 +236,64 @@ def _compute_log_partition_function(self,
         # Sum (log-sum-exp) over all possible tags
         return self._log_sum_exp(log_prob, 1)  # (batch_size,)
 
-    def _viterbi_decode(self, emission: torch.FloatTensor) -> List[int]:
-        # emission: (seq_length, num_tags)
-        assert emission.size(1) == self.num_tags
+    def _viterbi_decode(self, emissions: torch.FloatTensor, mask: torch.ByteTensor) \
+            -> List[List[int]]:
+        # Get input sizes
+        max_sequence_length = emissions.shape[0]
+        minibatch_size = emissions.shape[1]
+        sequence_lengths = mask.long().sum(dim=0)
+
+        # emissions: (seq_length, batch_size, num_tags)
+        assert emissions.shape[2] == self.num_tags
 
-        seq_length = emission.size(0)
+        # list to store the decoded paths
+        best_tags_list = []
 
         # Start transition
-        viterbi_score = self.start_transitions.data + emission[0]
+        viterbi_score = []
+        viterbi_score.append(self.start_transitions.data + emissions[0])
         viterbi_path = []
-        # Here, viterbi_score has shape of (num_tags,) where value at index i stores
+
+        # Here, viterbi_score is a list of shapes of (num_tags,) where value at index i stores
         # the score of the best tag sequence so far that ends with tag i
         # viterbi_path saves where the best tags candidate transitioned from; this is used
         # when we trace back the best tag sequence
 
         # Viterbi algorithm recursive case: we compute the score of the best tag sequence
         # for every possible next tag
-        for i in range(1, seq_length):
+        for i in range(1, max_sequence_length):
             # Broadcast viterbi score for every possible next tag
-            broadcast_score = viterbi_score.view(-1, 1)
+            broadcast_score = viterbi_score[i - 1].view(minibatch_size, -1, 1)
             # Broadcast emission score for every possible current tag
-            broadcast_emission = emission[i].view(1, -1)
+            broadcast_emission = emissions[i].view(minibatch_size, 1, -1)
             # Compute the score matrix of shape (num_tags, num_tags) where each entry at
```

Comment:
I think the comment should be updated. The matrix shape is now `(batch_size, num_tags, num_tags)`, right?